src/theory/quantifiers/ematching/inst_strategy_e_matching.cpp

```diff
@@ -235,153 +235,85 @@ InstStrategyStatus InstStrategyAutoGenTriggers::process(Node f,
   return InstStrategyStatus::STATUS_UNKNOWN;
 }
 
-void InstStrategyAutoGenTriggers::generateTriggers( Node f ){
-  Trace("auto-gen-trigger-debug") << "Generate triggers for " << f << ", #var=" << f[0].getNumChildren() << "..." << std::endl;
+void InstStrategyAutoGenTriggers::generateTriggers(Node q)
+{
+  Trace("auto-gen-trigger-debug")
+      << "Generate triggers for " << q << ", #var=" << q[0].getNumChildren()
+      << "..." << std::endl;
 
   // first, generate the set of pattern terms
-  if (!generatePatternTerms(f))
+  if (!generatePatternTerms(q))
   {
     Trace("auto-gen-trigger-debug")
         << "...failed to generate pattern terms" << std::endl;
     return;
   }
-
-  // then, group them to make triggers
-  unsigned rmin = d_patTerms[0][f].empty() ? 1 : 0;
-  unsigned rmax = options().quantifiers.multiTriggerWhenSingle ? 1 : rmin;
-  for (unsigned r = rmin; r <= rmax; r++)
-  {
-    std::vector<Node> patTerms;
-    std::vector<Node>& ptc = d_patTerms[r][f];
-    for (const Node& p : ptc)
-    {
-      if (r == 1 || d_single_trigger_gen.find(p) == d_single_trigger_gen.end())
-      {
-        patTerms.push_back(p);
-      }
-    }
-    if (patTerms.empty())
-    {
-      continue;
-    }
-    Trace("auto-gen-trigger") << "Generate trigger for " << f << std::endl;
-    // sort terms based on relevance
+  bool alreadyMadeTriggers = d_madeTriggers.find(q) != d_madeTriggers.end();
```

Comment:
Could clarify that already having made triggers for the formula will not prevent you from doing more stuff here.